polar/src/vm.rs

```diff
@@ -1986,21 +1989,26 @@ impl PolarVirtualMachine {
         }
     }
 
+    fn set_error_context(
+        &self,
+        term: &Term,
+        error: impl Into<error::PolarError>,
+    ) -> error::PolarError {
+        let source = self.source(term);
+        if source.is_none() {
```

Comment:
This could just be an expect, right?

Reply:
This was a placeholder :) Caught it in a follow up commit.